File: fintrack-backend/src/main/java/com/fintrack/fintrackbackend/dto/DashboardSummaryResponse.java
package com.fintrack.fintrackbackend.dto;

import com.fintrack.fintrackbackend.model.Transaction;

import java.math.BigDecimal;
import java.util.List;

public class DashboardSummaryResponse {

    private BigDecimal currentBalance;
    private BigDecimal monthlyIncome;
    private BigDecimal monthlyExpense;
    private List<Transaction> recentTransactions;

    public DashboardSummaryResponse(BigDecimal currentBalance,
                                    BigDecimal monthlyIncome,
                                    BigDecimal monthlyExpense,
                                    List<Transaction> recentTransactions) {
        this.currentBalance = currentBalance;
        this.monthlyIncome = monthlyIncome;
        this.monthlyExpense = monthlyExpense;
        this.recentTransactions = recentTransactions;
    }

    public BigDecimal getCurrentBalance() {
        return currentBalance;
    }

    public BigDecimal getMonthlyIncome() {
        return monthlyIncome;
    }

    public BigDecimal getMonthlyExpense() {
        return monthlyExpense;
    }

    public List<Transaction> getRecentTransactions() {
        return recentTransactions;
    }
}

File: frontend/app/src/main/java/pe/fintrack/mobile/ui/theme/data/model/DashboardSummaryResponse.kt
package pe.fintrack.mobile.ui.theme.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import pe.fintrack.mobile.ui.theme.data.Transaction
import java.math.BigDecimal

// Nota: Renombrado para coincidir con el DTO del backend
@JsonClass(generateAdapter = true)
data class DashboardSummaryResponse(
    @Json(name = "currentBalance") val currentBalance: BigDecimal,
    @Json(name = "monthlyIncome") val monthlyIncome: BigDecimal,
    @Json(name = "monthlyExpense") val monthlyExpense: BigDecimal,
    @Json(name = "recentTransactions") val recentTransactions: List<Transaction>
)